src/App/mermaidData/pullApiData.js

```diff
@@ -86,13 +88,29 @@ export const pullApiData = async ({
           const deletes = apiData[apiDataType]?.deletes ?? []
           const deleteIds = deletes.map(({ id }) => id)
           const error = apiData[apiDataType]?.error
-          const errorIds = error && [401, 403].includes(error.code) ? error.record_ids : []
-
-          // Use Set to remove duplicates
-          const bulkDeleteIds = Array.from(new Set([...deleteIds, ...errorIds]))
+          // it is implied that we will only get 401 and 403 sync errors for project-related tables
+          // those responses will include an array of ids to wipe out everything associated with the project
+          // causing the 401 or 403
+          const shouldDeleteErrorCodeAssociatedRecordsToo = error && [401, 403].includes(error.code)
+          const errorCodeAssociatedIds = shouldDeleteErrorCodeAssociatedRecordsToo
+            ? error.record_ids
+            : []
+          const isDataTypeProjectAssociated = getIsDataTypeProjectAssociated(apiDataType)
+
+          const bulkDeleteIdsWithNoDuplicates = Array.from(
+            new Set([...deleteIds, ...errorCodeAssociatedIds]),
+          )
```

Comment:
just renamed a var here to remove the need for a comment

Reply:
shouldDeleteErrorCodeAssociatedRecordsToo is a bit confusing for me

what about `deleteErrorCodeAssociatedRecords`

Reply:
its a boolean, but I get your point... hmm

Reply:
ah ok, what about isDeleteErrorCodeAssociatedRecords

Reply:
maybe `hasDeleteErrorCodeAssociatedRecords`...

Reply:
that works!